ansible/roles/install-microshift/tasks/main.yml

```diff
@@ -0,0 +1,237 @@
+  ansible.posix.firewalld:
+    source: 10.42.0.0/16
+    state: enabled
+    immediate: yes
+    permanent: yes
+    zone: trusted
+
+- name: permit traffic in public zone for services
+  ansible.posix.firewalld:
+    service: "{{ item }}"
+    state: enabled
+    immediate: yes
+    permanent: yes
+    zone: public
+  with_items: "{{ firewall_services }}"
+
+- name: permit traffic in public zone for ports
+  ansible.posix.firewalld:
+    port: "{{ item }}"
+    state: enabled
+    immediate: yes
+    permanent: yes
+    zone: public
+  with_items: "{{ firewall_ports }}"
+
+- name: upgrade all packages
+  ansible.builtin.dnf:
+    name: "*"
+    state: latest
+  notify: reboot machine
+
+- name: flush handlers
+  ansible.builtin.meta: flush_handlers
+
+- name: record initial disk space
+  vars:
+    filename: disk0.txt
+  include_tasks: disk.yml
+  loop: "{{ du_dirs }}"
+
+- name: enable microshift copr repo
+  community.general.copr:
+    name: '@redhat-et/microshift'
+    state: enabled
+
+- name: enable cri-o module
+  ansible.builtin.command: dnf module enable -y cri-o
+
+- name: install microshift package
+  ansible.builtin.dnf:
+    name:
+    - microshift
+    state: present
+    update_cache: true
+
+- name: pbench recording start
+  block:
+  - name: source pbench-agent & register-tool-set
+    ansible.builtin.shell: source /etc/profile.d/pbench-agent.sh && pbench-register-tool-set
+  
+  - name: start pbench-user-benchmark recording
+    ansible.builtin.shell: source /etc/profile.d/pbench-agent.sh && pbench-user-benchmark --config=microshift -- sleep {{ pbench_record_duration }}
+    async: "{{ pbench_record_duration|int * 2 }}"
+    poll: 0
+    register: pbench_user_benchmark_result
+  
+  - name: Pause for 60 seconds to gather steady state for pbench tool recording
+    ansible.builtin.pause:
+      seconds: 60
+  when: install_pbench | bool
+  environment:
+    PBENCH_TOOL_DATA_SINK: localhost:8081
+
+- name: record post-install disk utilization
+  vars:
+    filename: disk1.txt
+  include_tasks: disk.yml
+  loop: "{{ du_dirs }}"
+
+- name: force update of current timestamp
+  ansible.builtin.setup:
+    filter:
+      - 'ansible_date_time'
+
+- name: set start_time
+  ansible.builtin.set_fact:
+    start_time: "{{ ansible_date_time.iso8601[:19] }}"
+
+- name: start microshift service
+  ansible.builtin.systemd:
+    name: microshift
+    state: started
+    enabled: no
+
+- name: restart cadvisor to pick up new containers
+  ansible.builtin.systemd:
+    name: cadvisor
+    state: restarted
+
+- name: wait for kubeconfig to exist on microshift node
+  ansible.builtin.stat:
+    path: "{{ remote_kubeconfig_path }}"
+  register: remote_kubeconfig
+  until: remote_kubeconfig.stat.exists
+  retries: 30
+  delay: 10
+
+- name: wait for pods running
+  ansible.builtin.shell: kubectl get po -A --no-headers | grep Running | wc -l
+  retries: 60
+  delay: 5
+  register: result
+  until: result.stdout == "6"
```

Comment:
Instead of counting Running pods, we may want to "grep -v Running" and wait for 0. This should be more robust when pods are added.

Reply:
If microshift had a readiness check API endpoint, would we count those core pods as part of the answer it should return?

Reply:
I've changed the check as you've suggested @ggiguash.
@dhellmann I'm not sure if you're asking me or Gregory, but I assume a readiness check would count the core pods?

Is there any chance of a race where some pods are not yet created but the first ones are in Running state?

Reply:
The change broke the check, as the query returns "No resources found" for some time until some pods are created.
Any suggestions on what it should be if we don't like the pod running count check that I had before?

Reply:
Since this is a shell command we're running, we should be able to add a "sleep 1" loop on "No resource found" condition and then proceeding to the next stage once we have at least one pod.

Reply:
Not really a fan of the current wait situation, but we can discuss this further for future changes.